modules/cnf-troubleshooting-numa-aware-workloads.adoc

```diff
@@ -0,0 +1,176 @@
+// Module included in the following assemblies:
+//
+// *scalability_and_performance/cnf-numa-aware-scheduling.adoc
+
+:_content-type: PROCEDURE
+[id="cnf-troubleshooting-numa-aware-workloads_{context}"]
+= Troubleshooting NUMA-aware scheduling
+
+To troubleshoot common problems with NUMA-aware pod scheduling, perform the following steps.
+
+.Prerequisites
+
+* Install the {product-title} CLI (`oc`).
+
+* Log in as a user with cluster-admin privileges.
+
+* Install the NUMA Resources Operator and deploy the NUMA-aware secondary scheduler.
+
+.Procedure
+
+. Verify that the `noderesourcetopologies` CRD is deployed in the cluster:
++
+[source,terminal]
+----
+$ oc get crd | grep noderesourcetopologies
+----
++
+.Example output
+[source,terminal]
+----
+NAME                                                              CREATED AT
+noderesourcetopologies.topology.node.k8s.io                       2022-01-18T08:28:06Z
+----
+
+. Check that the NUMA-aware scheduler name matches the name specified in your NUMA-aware workloads:
++
+[source,terminal]
+----
+$ oc get numaresourcesschedulers.nodetopology.openshift.io numaresourcesscheduler -o json | jq '.status.schedulerName'
+----
++
+.Example output
+[source,terminal]
+----
+topo-aware-scheduler
+----
+
+. Verify that NUMA-aware scheduable nodes have the `noderesourcetopologies` CR applied to them.
++
+[source,terminal]
+----
+$ oc get noderesourcetopologies.topology.node.k8s.io
+----
++
+.Example output
+[source,terminal]
+----
+NAME                    AGE
+compute-0.example.com   17h
+compute-1.example.com   17h
+----
++
+[NOTE]
+====
+The number of nodes should equal the number of worker nodes that are configured by the machine config pool (`mcp`) worker definition.
+====
+
+. Verify the NUMA zone granularity for all scheduable nodes:
++
+[source,terminal]
+----
+$ oc get noderesourcetopologies.topology.node.k8s.io -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: topology.node.k8s.io/v1alpha1
+  kind: NodeResourceTopology
+  metadata:
+    annotations:
+      k8stopoawareschedwg/rte-update: periodic
+    creationTimestamp: "2022-01-24T12:14:32Z"
+    generation: 11184
+    name: compute-0.example.com
+    resourceVersion: "17195693"
+    uid: ef04fcc8-8022-4e85-8ad0-033640584966
+  topologyPolicies:
+  - SingleNUMANodeContainerLevel
+  zones: <1>
+  - costs:
+    - name: node-0
+      value: 10
+    - name: node-1
+      value: 21
+    name: node-0
+    resources: <2>
+    - allocatable: "51"
+      available: "33"
+      capacity: "52"
+      name: cpu
+    - allocatable: "47984070656"
+      available: "47459782656"
+      capacity: "49162670080"
+      name: memory
```

Comment:
let's probably add a hugepages resource reflection like this under each numa:
under node-0s:
```
    Allocatable:  6442450944
    Available:    6442450944
    Capacity:     6442450944
    Name:         hugepages-1Gi
    Allocatable:  134217728
    Available:    134217728
    Capacity:     134217728
    Name:         hugepages-2Mi
```

under node-1s:

```
    Allocatable:  1073741824
    Available:    1073741824
    Capacity:     1073741824
    Name:         hugepages-1Gi
    Allocatable:  268435456
    Available:    268435456
    Capacity:     268435456
    Name:         hugepages-2Mi
```
please reach out to me if you need help setting them.